src/Sprite/OP2BmpLoader.cpp

```diff
@@ -33,7 +35,18 @@ void OP2BmpLoader::ExtractImage(std::size_t index, const std::string& filenameOu
 
 std::vector<Color> OP2BmpLoader::GetPalette(const ImageMeta& imageMeta)
 {
-	std::vector<Color> palette(artFile.palettes[imageMeta.paletteIndex].size());
+	std::vector<Color> palette;
+
+	if (imageMeta.IsShadow())
+	{
+		// Shadow graphic uses a 2 color palette
+		palette.resize(2);
+	}
+	else
+	{
+		palette.resize(artFile.palettes[imageMeta.paletteIndex].size());
+	}
```

Comment:
Err, this looks like it's resizing the original palette in a "Get" method. This is rather unexpected.

Reply:
palette is defined in this function before being passed out. I don't think it is resizing the original palette from artFile, just setting the proper size for the new image's palette unless I missed something.

Maybe the function name could be changed to copyImagePaletteFromArtFile.

Reply:
Ahh, right, I read that wrong. Was probably too tired when I reviewed it.

This has got me curious though. Does the game really use just the first 2 colors from a full sized palette? Is this this an implementation detail for this particular project.

Reply:
The change is based on @TechCor8's port to C# and reimplementing those changes on OP2Helper. I haven't personally tested yet.